src/scripts/main.js

```diff
@@ -1 +1,94 @@
 'use strict';
+
+const firstPromise = new Promise((resolve, reject) => {
+  const timeoutId = setTimeout(() => {
+    reject(new Error('First promise was rejected'));
+  }, 3000);
+
+  document.addEventListener('click', (e) => {
+    if (e.button === 0) {
+      clearTimeout(timeoutId);
+      resolve('First promise was resolved');
+    }
+  });
+});
+
+const secondPromise = new Promise((resolve) => {
+  let isResolved = false;
+
+  const resolveOnce = (message) => {
+    if (!isResolved) {
+      isResolved = true;
+      resolve(message);
+    }
+  };
+
+  document.addEventListener('click', (e) => {
+    if (e.button === 0) {
+      resolveOnce('Second promise was resolved');
```

Comment:
Consider removing the click event listener after resolving the promise to prevent potential memory leaks.